qctoolkit/serialization.py

```diff
@@ -351,18 +352,21 @@ def __init__(self, identifier: Optional[str]=None, registration: weakref.WeakVal
         """
         super().__init__()
 
-        if registration is None:
-            registration = default_pulse_registration
+        if registry is None:
+            registry = default_pulse_registry
 
         if identifier == '':
             raise ValueError("Identifier must not be empty.")
         self.__identifier = identifier
 
-        if identifier and registration:
-            if identifier in registration:
+        if identifier and registry is not None:
+            # trigger garbage collection to "simulate" RAII
+            gc.collect(2)
```

Comment:
This increases the time required to run all tests on my machine by a factor of ~3 (~20 to ~60 secs). Why exactly is this necessary?

Reply:
Because garbage collection is not deterministic and there might me a pulse with the same name that has no references left but isn't collected yet.